File: app/schemas/veiculo.py
from datetime import datetime
from typing import List

from app.enums import VeiculoImagemStatus, VeiculoStatus
from core.schemas import BaseSchema


class VeiculoHistorico(BaseSchema):
    valor: float
    quilometragem: int
    descricao: str
    datahora: datetime
    veiculo_id: int

    class Config:
        orm_mode = True


class VeiculoImagem(BaseSchema):
    url: str
    veiculo_id: int
    status: VeiculoImagemStatus

    class Config:
        orm_mode = True


class Veiculo(BaseSchema):
    marca: str
    modelo: str
    ano: int
    url: str
    titulo: str
    site: str
    favorito: bool
    status: VeiculoStatus
    # historicos: List[VeiculoHistorico] = []
    # imagens: List[VeiculoImagem] = []

    class Config:
        orm_mode = True

File: static/js/components/Veiculo.js
export default function Site(site) {
    var _a;
    console.log(site);
    return (_a = site.veiculos) === null || _a === void 0 ? void 0 : _a.map((veiculo) => `<p><a href="${veiculo.url}" target="_blank"> ${veiculo.titulo}</a></p>`).join("");
}
